test/dist_autograd_test.py

```diff
@@ -99,19 +99,25 @@ def my_py_nested_call(t1, t2, dst, world_size, hops):
 # nodes. This helper allows timeout_seconds for those RPCs to be completed, and
 # ensures that all the contexts have been cleaned up in that timeframe.any
 def _all_contexts_cleaned_up(timeout_seconds=10):
+    assert dist.is_initialized(), "ProcessGroup must be initialized."
```

Comment:
Hmm, I think its better for the caller test methods to call `barrier()` on their own, as in some cases, the barrier is not necessary (see `test_context_cleanup_many_workers` and `test_context_cleanup_tensor_no_grad`). I'm not sure if we'd want to add `dist.barrier()` calls unless it is strictly necessary for the test

Reply:
@rohan-varma Moved `barrier()` back to the 3 tests where I noticed flakiness.